### PR TITLE
LibGfx: Implement curve parsing for lutAToBType and lutBToAType

### PR DESCRIPTION
**Pull request by nico**
With this, we parse all data of all permitted tag types of all
required tags in all ICC profile types :^)

### CODE REVIEW

**Comment by nico** [comment]
Looks fairly underwhelming so far:

```
% Build/lagom/icc ~/src/jpegfiles/stress.jpeg
[...]
AToB0Tag ('A2B0'): type 'mAB ', offset 400, size 1968
    3 input channels, 3 output channels
    a curves: 3 curves
    color lookup table: 11 x 5 x 5 grid points, 825 u16 entries
    m curves: 3 curves
    e = [ 1, 0, 0, 0,
          0, 1, 0, 0,
          0, 0, 1, 0 ]
    b curves: 3 curves
BToA0Tag ('B2A0'): type 'mBA ', offset 2368, size 1968
    3 input channels, 3 output channels
    b curves: 3 curves
    e = [ 1, 0, 0, 0,
          0, 1, 0, 0,
          0, 0, 1, 0 ]
    m curves: 3 curves
    color lookup table: 11 x 5 x 5 grid points, 825 u16 entries
    a curves: 3 curves
```

**Comment by nico** [comment]
(With stress.jpg being https://littlecms.com/img/blog/stress.jpg)

File: Userland/Libraries/LibGfx/ICC/Profile.cpp [hunk 1]
```diff
@@ -558,7 +558,9 @@ ErrorOr<void> Profile::read_header(ReadonlyBytes bytes)
 
 ErrorOr<NonnullRefPtr<TagData>> Profile::read_tag(ReadonlyBytes bytes, u32 offset_to_beginning_of_tag_data_element, u32 size_of_tag_data_element)
 {
-    // FIXME: Check 4-byte alignment on incoming offset.
```
Comment (linusg) [inline Userland/Libraries/LibGfx/ICC/Profile.cpp]: Seems silly to have a previous commit adding this FIXME in the same PR, please drop that.

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibGfx/ICC/Profile.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp]: Done. (Originally I thought that commit would be in a separate PR, but then I had a bonus 5 minutes.)

Comment (linusg) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp]: I think that rebase didn't quite work out - `LibGfx: Add a FIXME to ICC::Profile` is still there, with the removal being in the following commit :)

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp]: What about now?

**Pull request merged**